### PR TITLE
ath79: add support of led-running to DIR-825b1

### PR DESCRIPTION
**Pull request by hanipouspilot**
The code was taken from lantiq platform

Signed-off-by: Dmitry Tunin <hanipouspilot@gmail.com>

### CODE REVIEW

File: target/linux/ath79/base-files/etc/diag.sh [hunk 1]
```diff
@@ -16,7 +17,11 @@ set_state() {
 		status_led_blink_preinit_regular
 		;;
 	done)
-		status_led_on
+		status_led_off
+		[ -n "$running" ] && {
+			status_led="$running"
+			status_led_on
+		}
```
Comment (981213) [inline target/linux/ath79/base-files/etc/diag.sh]: It seemed that you'll turn off status led on device without a running led?

Comment (hanipouspilot) [reply target/linux/ath79/base-files/etc/diag.sh]: If there is no running, it will be on again.

Comment (hanipouspilot) [reply target/linux/ath79/base-files/etc/diag.sh]: Oops. You are right.

**Comment by mkresin** [comment]
May I ask for a different implementation of the functionality.

At the moment we have three different implementations of the script:

- the original implementation in lantiq
- an extended version with support for an upgrade led in apm821xx, ipq40xx and ipq806x
- a stripped down version supporting only a (different named) status led in ramips and ath79

I would like to see the naming of the LEDs and the implementations consolidated across the targets.

Since it is a boring job, I'm not sure if you are really interested. The bare minimum I would like to see here, is the use of the diag.sh version used by apm821xx, ipq40xx and ipq806x. Which means, in each dts we need the following led aliases:

- led-boot
- led-failsafe
- led-running
- led-upgrade

Pointing the same LED to not change anything.

**Comment by hanipouspilot** [comment]
@mkresin Why do we need the aliases in EACH dts file?
Most of devices are using one led for all these tasks. In rare cases we have multiple.
Am not familiar with all devices already ported to ath79, but most of them don't need this.
This may be added in case it is really needed.

**Comment by mkresin** [comment]
It is as simple as [diag.sh](https://github.com/openwrt/openwrt/blob/master/target/linux/apm821xx/base-files/etc/diag.sh) from apm821xx, ipq40xx and ipq806x expects each led defined.

If you have a look at the failsafe case for example, the status_led is set to "$failsafe" and the failsafe blink pattern is started for the led stored in the status_led variable.

**Comment by hanipouspilot** [comment]
@mkresin Right. But do we really need this in ath79? These are older devices mostly without so many fancy leds. It is not too much work to add some lines to dts files and copy the diag.sh, but in 99% it will be useless.

**Comment by hanipouspilot** [comment]
My case was the first one when we needed more than one. Why not leave it as is until someone asks for more?

**Comment by mkresin** [comment]
Well, I would like to see it consolidated instead of adding yet another way of doing the same.

Have a look at the ar9330_glinet_ar150.dts for example. It hijacks the wireless led for boot status indication. In such a case I would define everything except "led-running" to "free" the led at the end of the boot process. If you have a look at ar71xx, you usually find a ucidef_set_led_default to do the same.

Your board already requires a boot and a running led. Might make sense to define the boot led as failsafe as well. This way it is easy to distinguish if you are in normal boot or in failsafe.

And soon there isn't much different to the other targets using devicetree based status leds. Sure it isn't used in ar71xx, but it might be a chicken/egg problem.

The qca9563_phicomm_k2t.dts has a three different power colours and might can use something like the lantiq/bt homehub 5a.

**Comment by hanipouspilot** [comment]
I really don't like this idea to put too much, when it is not really needed.
Regarding the failsafe, we already have 

```
failsafe)
		status_led_blink_failsafe
		;;
```
Status led blinks fast.
We have a simple and clean diag.sh. It doesn't meen to do it "a different way".
My points are:

1. We don't have to add multiple leds until it is required.
2. Adding useless aliases to all dts files is not athe best way of doing it.
Why not check existence of aliases in diag.sh?

I see your point that if we have a device that requires all that, it would me much cleaner to have it.
But why don't we wait till that "if" is "true"?

**Comment by mkresin** [comment]
> I really don't like this idea to put too much, when it is not really needed.

As said, chicken egg. It wasn't "needed" for the other targets as well. But as soon as it was added it was used.

> I see your point that if we have a device that requires all that, it would me much cleaner to have it.
But why don't we wait till that "if" is "true"?

Because I try to keep the implementations across the targets identical. It's a paint to fix a bug for one target and try to port the fix to another target which has a slightly different implementation.

I guess we should agree to disagree here. It isn't worth to "fight" about this detail.

**Comment by mkresin** [comment]
> And another example. Do we need to turn status_led off on running as with some other targets? It should be on be on by default.

Not if a led is misused for boot status indication. Examples would be a wifi or wps led. As written above, these are usually turned of after boot with a ucidef_set_led_default like [here](https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/base-files/etc/board.d/01_leds#L38), [here](https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/base-files/etc/board.d/01_leds#L164), [here](https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/base-files/etc/board.d/01_leds#L474) ...

Trust me, it will be required and it totally make sense to add the possible to do so right from the beginning instead of changing the code later again. We have such a code in the other targets for a reason :-).

**Comment by hanipouspilot** [comment]
OK. Tommorrow I'll think about it again, and probably do the change.

**Comment by hanipouspilot** [comment]
@mkresin But I want to see normal status led on, not off. I need to give some thought.

**Comment by mkresin** [comment]
> But I want to see normal status led on, not off. I need to give some thought.

The status led will be on, if defined as led-running.

**Comment by hanipouspilot** [comment]
That's another point in your favor. ;-) It is really easier to do it this way.
пн, 13 авг. 2018 г. в 19:01, Mathias Kresin <notifications@github.com>:

**Comment by mkresin** [comment]
Superseded by #1286

**Pull request closed without merging**